homeassistant/config.py

```diff
@@ -86,19 +86,22 @@
 """
 
 
-def _valid_customize(value):
-    """Config validator for customize."""
-    if not isinstance(value, dict):
-        raise vol.Invalid('Expected dictionary')
+CUSTOMIZE_SCHEMA_ENTRY = vol.All(
+    cv.has_at_least_one_key('entity_id', 'entity_id_glob'),
+    cv.ordered_dict(cv.match_all, cv.string))
 
-    for key, val in value.items():
-        if not valid_entity_id(key):
-            raise vol.Invalid('Invalid entity ID: {}'.format(key))
 
-        if not isinstance(val, dict):
-            raise vol.Invalid('Value of {} is not a dictionary'.format(key))
+def _dict_to_list(inp: Any) -> List:
+    if not isinstance(inp, dict):
+        return cv.ensure_list(inp)
+    if 'entity_id' in inp:
+        return [inp]  # sigle entry
+    res = []
 
-    return value
+    for key, val in cv.ordered_dict(OrderedDict)(inp).items():
```

Comment:
Agreed, but `OrderedDict` is not a voluptuous schema. (`vol.Schema(OrderedDict)` might do what you intend to)

If you follow the link in my previous post you will also see it will *never* be an OrderedDict being passsed into this schema, but always a normal dict with a single key being added (in order) to an OrderedDict.